java/src/jmri/InstanceManagerAutoInitialize.java

```diff
@@ -0,0 +1,23 @@
+package jmri;
+
+/**
+ * Provide a hint to the {@link jmri.InstanceManager} that this object needs
+ * have additional initialization performed after it is made available by the
+ * InstanceManager. This allows two classes that have circular dependencies on
+ * being able to get the default instance of each other to be managed
+ * successfully.
+ * <p>
+ * Note: the need to have a class implement this probably is indicative of other
+ * design issues in the implementing class.
+ *
+ * @author Randall Wood Copyright 2017
+ */
+public interface InstanceManagerAutoInitialize {
```

Comment:
Below is I think the safe example, with rules:
- Don't ask InstanceManager for references in your constructor
- You can ask for references in initialize(), but you can't assume initialization of the referneced object is complete

![instancesequence2](https://user-images.githubusercontent.com/2774117/29006561-40d3c566-7aa7-11e7-8a00-70535bbd648d.png)
[InstanceSequence2.txt](https://github.com/JMRI/JMRI/files/1203395/InstanceSequence2.txt)